be/src/storage/lake/meta_file.cpp

```diff
@@ -56,12 +56,15 @@ void MetaFileBuilder::append_delvec(const DelVectorPtr& delvec, uint32_t segment
 
 void MetaFileBuilder::apply_opwrite(const TxnLogPB_OpWrite& op_write,
                                     const std::map<int, std::string>& replace_segments,
-                                    const std::vector<std::string>& orphan_files) {
+                                    const std::vector<std::string>& orphan_files,
+                                    std::map<std::string, uint64_t>& replace_segments_file_size) {
     auto rowset = _tablet_meta->add_rowsets();
     rowset->CopyFrom(op_write.rowset());
     for (const auto& replace_seg : replace_segments) {
         // when handle partial update, replace old segments with new rewrite segments
         rowset->set_segments(replace_seg.first, replace_seg.second);
+        // update new rewrite segments size
```

Comment:
if `rowset` is from old version, if you call set_segment_size(2 /* index */, 100); wiil index 0 and 1 be initialized to 0?

Reply:
You're right,  `set_segment_index` will index out of bound